lib/device-log/ios-log.js

```diff
@@ -249,7 +252,25 @@ class IOSLog {
   async getAllLogs () {
     return this.logs;
   }
+
+  /**
+   * Get 'idevicesyslog' subprocess for a given udid.
+   * 
+   * This function caches and re-uses instances so that there's
+   * just one instance per device (cache by UDID)
+   */
+  static async _getIDeviceSysLogProcess (cmd, args, env, udid) {
```

Comment:
why is this an async fn?

Reply:
Just a habit. I thought we had a rule to catch this.

Reply:
We _will_ have a rule. It requires updating eslint on all packages. So we do not at the moment. :|

Reply:
Good thing you spotted this @jlipps. I had assigned `proc` to the promise returned by `_getIDeviceSysLogProcess` instead of the actual `SubProcess`.